test/functional/Java8andUp/src/org/openj9/test/attachAPI/TestJstack.java

```diff
@@ -0,0 +1,231 @@
+ *******************************************************************************/
+package org.openj9.test.attachAPI;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.lang.Thread.State;
+import java.lang.management.ThreadInfo;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+
+import org.openj9.test.util.StringPrintStream;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+import openj9.tools.attach.diagnostics.info.JvmThreadInfo;
+import openj9.tools.attach.diagnostics.info.ThreadGroupInfo;
+
+@Test(groups = { "level.extended" })
+public class TestJstack extends AttachApiTest {
+
+	private static final String JSTACK_COMMAND = "jstack"; //$NON-NLS-1$
+	Object syncObject = new Object();
+
+	@Test
+	public void testSingleThread() throws IOException {
+		try {
+			Thread testThread = new Thread(this::method1);
+			testThread.start();
+			while (testThread.getState() != State.WAITING) {
+				try {
+					Thread.sleep(1);
+				} catch (InterruptedException e) {
+					/* ignore */
+				}
+			}
+			final long[] threadIds = new long[] {testThread.getId()};
+			Properties props = ThreadGroupInfo.getThreadInfoProperties(threadIds).toProperties();
+			logProperties(props);
+			JvmThreadInfo jvmThrInfo = new JvmThreadInfo(props, 0);
+			ThreadInfo thrInfo = jvmThrInfo.getThreadInfo();
+			log(thrInfo.toString());
+			compareThreadInfo(thrInfo, testThread);
+		} finally {
+			synchronized (syncObject) {
+				syncObject.notifyAll();	
+			}
+		}
+	}
+
+	@Test
+	/* basic sanity: fetch and print the thread information */
+	public void testAllThreads() throws IOException {
+		Properties props = ThreadGroupInfo.getThreadInfoProperties().toProperties();
+		logProperties(props);
+		ThreadGroupInfo grpInfo = new ThreadGroupInfo(props, false);
+		log(grpInfo.toString());
+	}
+
+	@Test
+	public void testMonitors() throws IOException {
+		synchronized (syncObject) {
+			Thread testThread = new Thread(new Runnable() {
+				@Override
+				public void run() {
+					syncMethod();
+				}
+			}, "testMonitorsThread"); //$NON-NLS-1$
+			testThread.start();
+			while (testThread.getState() != State.BLOCKED) {
+				try {
+					Thread.sleep(1);
+				} catch (InterruptedException e) {
+					/* ignore */
+				}
+			}
+			Properties props = ThreadGroupInfo.getThreadInfoProperties().toProperties();
+			logProperties(props);
+			ThreadGroupInfo grpInfo = new ThreadGroupInfo(props, false);
+			final String jstackOutput = grpInfo.toString();
+			log("jstack output:\n"+jstackOutput); //$NON-NLS-1$
```

Comment:
formatting: space around binary operators

Reply:
Fixed here and elsewhere in the file.

Reply:
To be clear, such suggestions are meant to apply to the pull request as a whole.